docs/architecture/ADR-007-bad-debt.md

```diff
@@ -0,0 +1,92 @@
+When fully liquidated, undercollateralized debt will result in an account with nonzero total borrows but zero collateral, thus no incentive for repayment.
+
+It is in the interests of the overall system to repay such _bad debt_ using reserves, in order to prevent it accruing interest forever and damaging the health of the lending pool.
+
+## Alternatives
+
+### Repayment Eligibility Condition
+
+One consideration to make is whether borrowers should be eligible for bad debt repayment whenever `Collateral Value < Borrow Value`, or if eligibility should be delayed until `Collateral Value == 0`.
+
+In the former case, the system would also confiscate any remaining collateral, as though it were a liquidator, and would have to process the collateral in a way of our choosing (e.g. exchange for base assets, and add them to reserves).
+
+In the latter case, liquidators would first have liquidate the portion of the debt that can be exchanged for collateral, reducing the total burden on reserves when the borrow is being repaid.
+
+Also in the latter case, if liquidators do not (for whatever reason) liquidate until the undercollateralized account's collateral is zero, its borrows would continue to accrue interest and remain ineligible for bad debt repayment.
+This should not be a common case, as long as liquidation incentives exist, but might happen if liquidators are sparse, or if the remaining collateral is so small that it is not worth the gas fee to retrieve.
+
+### Checking for Repayment Eligibility
+
+A second consideration is when to check for borrowers eligible for bad debt repayment. The two likely options are immediately during `LiquidateBorrow`, or during `EndBlock` every `InterestEpoch`. A third option might be to check periodically, but on a separately controlled interval `BadDebtRepayEpoch`.
+
+The advantage of an immediate check during `LiquidateBorrow` is that only the borrow being liquidated needs to be checked for eligibility, instead of periodically iterating over all borrows. Additioanlly, `CollateralValue` has already been calculated in that function.
+
+However, there is an edge case (reserve exhaustion) where borrows eligible for reserve-driven-repayment during their final liquidation, cannot be fully repaid at that moment.
+If using an immediate check during `LiquidateBorrow`, there would be no future liquidation against that address to trigger bad debt repayment after reserves recovered - the debt would accrue interest indefinitely.
+
+Also note that the fist decision on the condition for bad debt repayment eligibility changes the effectiveness of these options:
+
+- A condition of `Collateral Value == 0` favors checks during `LiquidateBorrow`. because liquidation is the only action on undercollateralized accounts that is allowed to reduce collateral to zero.
```

Comment:
```suggestion
- A condition of `Collateral Value == 0` favors checks during `LiquidateBorrow`, because liquidation is the only action on under-collateralized accounts that is allowed to reduce collateral to zero.
```